pkg/member/member_control.go

```diff
@@ -136,6 +144,34 @@ func (m *memberControl) AddMemberAsLearner(ctx context.Context) error {
 	return nil
 }
 
+// AddLearnerWithRetry add a new member as a learner with exponential backoff.
+func (m *memberControl) AddLearnerWithRetry(ctx context.Context, retrySteps int, dataDir string) error {
+	backoff := miscellaneous.CreateBackoff(RetryPeriod, retrySteps)
+
+	if err := retry.OnError(backoff, func(err error) bool {
+		return err != nil
+	}, func() error {
+		// Additional safety check before adding a learner
+		if _, err := os.Stat(dataDir); err == nil {
+			if err := os.RemoveAll(filepath.Join(dataDir)); err != nil {
+				return fmt.Errorf("failed to remove directory %s with err: %v", dataDir, err)
+			}
+		} else if !os.IsNotExist(err) {
```

Comment:
I might've missed something, but we aren't handling the case here where `os.Stat` returns an error that is not `os.IsNotExist`. We're just continuing onto `AddMemberAsLearner`

Reply:
please check this https://github.com/gardener/etcd-backup-restore/pull/617#discussion_r1178837127

Reply:
Okay, thank you